app/src/main/java/com/dessalines/thumbkey/ui/components/keyboard/KeyboardScreen.kt

```diff
@@ -150,8 +154,22 @@ fun KeyboardScreen(
     val backdropColor = MaterialTheme.colorScheme.background
     val backdropPadding = 6.dp
     val keyPadding = settings?.keyPadding ?: DEFAULT_KEY_PADDING
-    val legendHeight = settings?.keySize ?: DEFAULT_KEY_SIZE
-    val legendWidth = settings?.keyWidth ?: legendHeight
+    val autoSizeKeys = (settings?.autoSizeKeys ?: DEFAULT_AUTO_SIZE_KEYS).toBool()
```

Comment:
I'm guessing something in this block below is causing your stackoverflow.

Reply:
I don't see anything here *per se*. Some more review of current logs (after addressing the other comments above so the log is different) does show something migration related:

```
E AndroidRuntime:        at com.dessalines.thumbkey.db.AppDbKt$MIGRATION_17_18$1.migrate(AppDb.kt:657)
```

Reply:
I don't see anything here... what exactly was your hunch about?

I just cleaned this up so the new function returns Int instead of Float and then running toInt on it anyway, but that doesn't change the type of `legentHeight`.

Reply:
The stackoverflow had to do with this line: com.dessalines.thumbkey.ui.components.keyboard.KeyboardScreenKt.KeyboardScreen(KeyboardScreen.kt:161

Reply:
Old news, my initial debug log above wasn't from the code currently pushed to this PR. Sorry for the confusion; I kept throwing spaghetti at the wall hoping something would stick. It turns out there were two problems at once and changing one thing at a time, then reverting and trying something else didn't get me over the finish line because both issues had to be fixed in a way that matched.

Anyhow I think I've finally squashed both bugs.